File: C++/lcm.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int lcm(int x, int y)
{
    int a = max(x, y);
    int b = min(x, y);
    
    if (a % b == 0)
        return(a);
    
    for (int i = a+1; i <= a*b; i++)
        if (i % a == 0 and i % b == 0)
            return(i);
}

int main()
{
    int a, b;
    cin >> a >> b;
    cout << lcm(a, b) << endl;
    return 0;
}

File: Nim/lcm.nim
import strutils

proc lcm(x : int, y : int): int =
  let a : int = max(x, y)
  let b : int = min(x, y)
  
  if a mod b == 0:
    return a
  
  for i in countup(a+1, a*b):
    if i mod a == 0 and i mod b == 0:
      return i

let x = parseInt(readline(stdin))
let y = parseInt(readline(stdin))

echo lcm(x, y)
